### ISSUE TITLE
Improve dashboard performance

### ISSUE THREAD

**Issue by leepeuker** [leepeuker/movary#394]
Currently when loading the dashboard we are fetching the data for all possible dashboard rows regardless of which rows are really shown in the frontend. 

- [ ] only fetch the data for visible rows

**Comment by JVT038** [comment]
Would this mean that the dashboard row data would only be loaded when the user unfolds / extends the row?

**Comment by leepeuker** [comment]
Yeah good question. I was thinking to still keep it in the backend, so only to remove the data for the rows wich are not rendered. 

We could move the data fetching into js frontend, but that would mean we have to render it in the frontend too. This would probably the the best thing to do, but requires the most change effort.

**Comment by JVT038** [comment]
It shouldn't be that difficult right?

Just add an event listener to check whether the user has extended the rows and if yes, add a loading spinner and make a POST request in the background to the backend, to fetch the data. When the data has arrived, remove the spinner and manipulate the DOM to inject the requested data.

**Comment by leepeuker** [comment]
I did not say that it is difficult, but it takes time

**Comment by leepeuker** [comment]
We have reported performance issues on the dashboard, so increase the prio of this issue
